app/controllers/CheckYourAnswersController.scala

```diff
@@ -108,43 +108,39 @@ class CheckYourAnswersController @Inject()(identify: IdentifierActionProvider,
+                                     (implicit request: Request[?]): Future[RegistrationResponse] = {
     answers.registrationResponse
       .map(x => Future.successful(x))
-      .getOrElse {
-        answers.get(BusinessTypePage).map {
-          case Individual | SoleTrader => IndividualWithoutId.build(answers).fold(
-            errors => Future.failed(Exception(s"Unable to build a registration request for an individual, path(s) missing: ${errors.toChain.toList.map(_.path).mkString(", ")}")),
-            request => registrationConnector.register(request)
-          )
-          case _ => OrganisationWithoutId.build(answers).fold(
-            errors => Future.failed(Exception(s"Unable to build a registration request for an organisation, path(s) missing: ${errors.toChain.toList.map(_.path).mkString(", ")}")),
-            request => registrationConnector.register(request)
-          )
-        }.getOrElse(Future.failed(Exception("Could not find an answer for BusinessType when trying to build a registration request")))
-      }
+      .getOrElse(RegistrationRequest.build(answers).fold(
+        errors => Future.failed(BuildRegistrationRequestFailure(errors)),
+        registrationRequest => registrationConnector.register(registrationRequest)
+      ))
+  }
 
   private def subscribe(safeId: String, answersWithRegistration: UserAnswers, originalAnswers: UserAnswers)
-                       (implicit request: Request[?]): Future[SubscriptionDetails] =
+                       (implicit request: Request[?]): Future[SubscriptionDetails] = {
     lazy val isAutoSubscription = originalAnswers.registrationResponse.isEmpty
     SubscriptionRequest.build(safeId, answersWithRegistration).fold(
-      errors => Future.failed(Exception(s"Unable to build a subscription request, path(s) missing: ${errors.toChain.toList.map(_.path).mkString(", ")}")),
-      request =>
-        subscriptionConnector
-          .subscribe(request)
-          .map { response =>
-            auditService.sendAudit(AuditEventModel(isAutoSubscription, answersWithRegistration.data, response))
-            SubscriptionDetails(response, request, answersWithRegistration, false)
-          }
-          .recover {
-            case error =>
-              auditService.sendAudit(AuditEventModel(isAutoSubscription, answersWithRegistration.data, error.asInstanceOf[SubscribeFailure]))
-              throw error
-          }
+      errors => Future.failed(BuildSubscriptionRequestFailure(errors)),
+      request => subscriptionConnector
+        .subscribe(request)
+        .map { response =>
+          auditService.sendAudit(AuditEventModel(isAutoSubscription, answersWithRegistration.data, response))
+          SubscriptionDetails(response, request, answersWithRegistration, false)
+        }.recover {
+          case error =>
+            auditService.sendAudit(AuditEventModel(isAutoSubscription, answersWithRegistration.data, error.asInstanceOf[SubscribeFailure]))
```

Comment:
Need a unit test for case error

Reply:
This is not a code I have written. It was missed before. We have many places like this.